### PR TITLE
Check for isTouched before opening search suggestions

### PR DESCRIPTION
**Pull request by ataker**
## Chromatic

https://dst2676-search-suggestions--65a6e2ed2314f7b8f98609d8.chromatic.com

## Description
Closes [Search Typeahead component does not close as expected when the component loads with a value prefilled #2676](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2676)

Do not open search suggestions if the search hasn't been touched.. This would happen if the search was filled on load.
To test:

1. Publish this web-components to Verdaccio
2. Switch vets-website to branch 17176-search-results-wc
3. In vets-website, set web-components to the build you published to Verdaccio
4. Test

## QA Checklist
- [x] Component maintains 1:1 parity with design mocks
- [x] Text is consistent with what's been provided in the mocks
- [x] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [x] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [x] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [x] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots
Before
https://github.com/department-of-veterans-affairs/component-library/assets/1413938/672b384c-5431-4d95-ab66-87219b639ab2

After
https://github.com/department-of-veterans-affairs/component-library/assets/1413938/851ea977-14c2-4277-86e3-1560424a70d3

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by powellkerry — COMMENTED** [review]
LGTM

**Review by powellkerry — APPROVED** [review]

File: packages/web-components/src/components/va-search-input/va-search-input.tsx [hunk 1]
```diff
@@ -439,7 +452,7 @@ export class VaSearchInput {
       return (
         <Host onBlur={handleBlur}>
           <form class={formClasses} role="search">
-            <label class="usa-sr-only" htmlFor="search-field">{label}</label>
+            <label class="usa-sr-only kokokok" htmlFor="search-field">{label}</label>
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-search-input/va-search-input.tsx:455]: Is "kokokok" just a leftover thing from testing?

File: packages/web-components/src/components/va-search-input/va-search-input.tsx [hunk 2]
```diff
@@ -384,6 +391,12 @@ export class VaSearchInput {
       .slice(0, 5)
       .sort()
       .map(suggestion => this.formatSuggestion(suggestion));
+
+    if(!this.isTouched) {
+      // only open the listbox after the search input has been touched
+      this.isListboxOpen = false;
+      return
+    }
     this.isListboxOpen = true;
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-search-input/va-search-input.tsx:400]: Just a nit here but would a ternary simplify this logic or would that make it less readable?

```js
this.isTouched ? this.isListBoxOpen = true : this.isListboxOpen = false;
```

File: packages/web-components/src/components/va-search-input/va-search-input.tsx (hunk 1 above)
Comment (ataker) [reply packages/web-components/src/components/va-search-input/va-search-input.tsx:455]: Whoops XD

**Review by jamigibbs — APPROVED** [review]
Looks great, Alex! Thank you for digging into this one! 👍

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**